packages/db/src/lib/execution.ts

```diff
@@ -178,21 +178,31 @@ export const getWorkflowNameIntersectFromGranuleIds = async (
   const numberOfGranules = granuleCumulusIdsArray.length;
   const { executions: executionsTable, granulesExecutions: granulesExecutionsTable } = TableNames;
 
-  const aggregatedWorkflowCounts = await knexOrTransaction(executionsTable)
-    .select('workflow_name')
-    .countDistinct('granule_cumulus_id')
+  const aggregatedWorkflowCounts: Array<{
+    workflow_name: string,
+    min: number
+  }> = await knexOrTransaction(
+    executionsTable
+  )
+    .select(['workflow_name']).min('timestamp')
```

Comment:
This update seems *great* with respect to addressing the units issue, did you check the query analysis to make sure min combined with this order/by and join is going to work ok at some reasonable scale?

Reply:
I have measured the behavior and scaling of the query with and without this extra min() bit and it doesn't add any appreciable overhead. there is no observable change, except in the case that there are many workflows all with the same name corresponding to one granule, an din that case it made a difference of ~1 second for 10000 workflows of the same name to 1 granule

Reply:
that is, having changed it to only request the min on 1 granule, since it only needs it in that case

Reply:
Adding that conditional *definitely* mitigates any concerns.    Good work, thanks for looking into that.